br/pkg/pdutil/pd_serial_test.go

```diff
@@ -239,11 +240,21 @@ func TestPauseSchedulersByKeyRange(t *testing.T) {
 
 	labelExpires := make(map[string]time.Time)
 
+	var (
+		mu      sync.Mutex
+		deleted bool
+	)
+
 	httpSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		mu.Lock()
+		defer mu.Unlock()
+		if deleted {
+			return
+		}
```

Comment:
Does this lock only protects the `deleted`? What about making the `deleted` an atomic value and do a CAS before processing a request:
```golang
if r.Method == http.MethodDelete && deleted.CAS(false, true) {
    ruleID := ....
    delete(....)
}
```

Just a recommendation to make code cleaner and more efficient from my perspective, but it seems this is not a high-frequent operation so the atomic alternative might get limited improvement though.

Reply:
The lock doesn't protect the `deleted` only. More importantly, it ensures the server only handles a request at a time.